src/v/iceberg/manifest.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "container/fragmented_vector.h"
+#include "iceberg/manifest_entry.h"
+#include "iceberg/partition.h"
+#include "iceberg/schema.h"
+
+namespace iceberg {
+
+enum class format_version : uint8_t {
+    v1,
+    v2,
+};
+
+enum class manifest_content_type {
+    data = 0,
+    deletes = 1,
+};
```

Comment:
why does one have an explicit underlying type, and the other has explicit values (which are also the default values specified by the standard). generally whenever an enum class has customization (either underlying tpye, or values) then a comment is useful because if it is a change that is needed to maintain compatibility with something that is serialized then it can help prevent mistakes later.